src/Nethermind/Nethermind.Blockchain/Find/BlockExtensions.cs

```diff
@@ -125,6 +125,36 @@ public static class BlockExtensions
             }
         }
 
+        public static IEnumerable<(long LogIndex, long TransactionLogIndex, TxReceipt Receipt, LogEntry Entry)> FindLogsWithReceipts(
+            this BlockHeader blockHeader,
+            TxReceipt[] receipts,
+            LogFilter logFilter,
+            FindOrder receiptFindOrder = FindOrder.Ascending, // iterating forwards, by default we are interested in all items in order of appearance 
+            FindOrder logsFindOrder = FindOrder.Ascending) // iterating forwards, by default we are interested in all items in order of appearance
+        {
+            if (logFilter.Matches(blockHeader.Bloom))
+            {
+                int transactionLogIndex = 0;
+                for (int i = 0; i < receipts.Length; i++)
+                {
+                    TxReceipt receipt = GetItemAt(receipts, i, receiptFindOrder);
+                    if (logFilter.Matches(receipt.Bloom))
+                    {
+                        int logIndex = 0;
+                        for (int j = 0; j < receipt.Logs.Length; j++)
```

Comment:
logIndex instead of j would be better -> this way you avoid copy paste issues which confuse i and j